app/ui-react/syndesis/src/modules/integrations/components/editor/choice/ChoiceStepPage.tsx

```diff
@@ -59,6 +60,19 @@ export class ChoiceStepPage extends React.Component<IChoiceStepPageProps> {
               <WithRouteData<IChoiceStepRouteParams, IChoiceStepRouteState>>
                 {(params, state, { history }) => {
                   const positionAsNumber = parseInt(params.position, 10);
+                  let dataShape = {} as DataShape;
+                  try {
+                    const prevStep = getPreviousIntegrationStepWithDataShape(
+                      state.integration,
+                      params.flowId,
+                      positionAsNumber
+                    );
+                    dataShape =
+                      prevStep!.action!.descriptor!.outputDataShape ||
+                      ({} as DataShape);
+                  } catch (err) {
+                    // ignore
+                  }
```

Comment:
Would it make more sense to maybe extract this block of logic out into a separate function in `utils.ts`?